src/Persistence/MassTransit.EntityFrameworkIntegration/Saga/EntityFrameworkSagaRepository.cs

```diff
@@ -82,6 +82,9 @@ void IProbeSite.Probe(ProbeContext context)
             using (DbContext dbContext = _sagaDbContextFactory())
             using(var transaction = dbContext.Database.BeginTransaction(_isolationLevel))
             {
+                var tableName = ((IObjectContextAdapter)dbContext).ObjectContext.CreateObjectSet<TSaga>().EntitySet.Name;

```

Comment:
I don't think this is the right line for this to happen. It would need to occur later, where the .Set() is happening, so that it replaces the existing read with a new one, right? Otherwise it's being read twice from SQL?